cl/scrapers/management/commands/merge_opinion_versions.py

```diff
@@ -0,0 +1,688 @@
+    "appellate_case_type_information",
+    "mdl_status",
+    "filepath_local",
+    ("date_blocked", min),
+    "blocked",
+]
+
+cluster_fields_to_merge = [
+    ("judges", merge_judge_names),
+    "date_filed",
+    "date_filed_is_approximate",
+    "case_name_short",
+    "case_name_full",
+    "scdb_id",
+    "scdb_decision_direction",
+    "scdb_votes_majority",
+    "scdb_votes_minority",
+    ("source", SOURCES.merge_sources),
+    "procedural_history",
+    "attorneys",
+    "nature_of_suit",
+    "posture",
+    "syllabus",
+    "blocked",
+    ("date_blocked", min),
+    "headnotes",
+    "cross_reference",
+    "correction",
+    "disposition",
+    "other_dates",
+    "summary",
+    "filepath_json_harvard",
+    "filepath_pdf_harvard",
+    "arguments",
+    "headmatter",
+]
+
+
+def get_query_from_url(url: str, url_filter: str) -> Q:
+    """Build an OR query with the alternative to http or https
+
+    :param url: the input url
+    :param url_filter: a string that identified a Django filter.
+        Currently supporting
+        - 'startswith'
+        - 'exact', meaning an equality filter
+
+    :return: a Q OR object
+    """
+    if url_filter not in ["exact", "startswith"]:
+        raise ValueError("`url_filter` must be in [exact, startswith]")
+
+    if "https" in url:
+        extra_query = url.replace("https", "http")
+    else:
+        extra_query = url.replace("http", "https")
+
+    if url_filter == "startswith":
+        return Q(download_url__startswith=url) | Q(
+            download_url__startswith=extra_query
+        )
+
+    return Q(download_url=url) | Q(download_url=extra_query)
+
+
+def clean_opinion_text(opinion: Opinion) -> str:
+    """Remove noise (HTML tags or extra whitespace) from an opinion's text
+
+    :param opinion: the opinion
+    :return: the cleaned text
+    """
+    if opinion.html:
+        return clean_text(opinion.html, ["html", "all_whitespace"])
+
+    return re.sub(r"\s+", " ", opinion.plain_text)
+
+
+def text_is_similar(text1: str, text2: str) -> bool:
+    """Check if the text from both opinions is the same or very similar
+
+    :param text1: a cleaned opinion's text
+    :param text2: another cleaned opinion's text
+    """
+    if text1 == text2:
+        return True
+
+    # a single character difference yields a 0.999 ratio
+    # a single word difference may reduce the similarity a few points,
+    # depending on how long the sequence is, and if it is an addition/removal
+    # or a correction. In general, a 0.9 MIN value should tolerate a difference
+    # of a few words between versions, which is what we expect
+    # Note that this in sensible to argument order, so we retry with inverted
+    # order if the first run fails
+    sm = SequenceMatcher(None, text1, text2)
+    ratio = sm.ratio()
+    if ratio < MIN_SEQUENCE_SIMILARITY:
+        return (
+            SequenceMatcher(None, text2, text1).ratio()
+            > MIN_SEQUENCE_SIMILARITY
```

Comment:
maybe >= instead of > to include 0.9 which is the min value? (the same for L318)